pyproject.toml

```diff
@@ -26,3 +26,82 @@ target-version = "py38"
 
 [tool.ruff.isort]
 known-first-party = ["vcr"]
+
+[build-system]
+requires = ["setuptools>=61.2"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "vcrpy"
+version = "5.1.0"
+authors = [{name = "Kevin McCarthy", email = "me@kevinmccarthy.org"}]
+license = {text = "MIT"}
+description = "Automatically mock your HTTP interactions to simplify and speed up testing"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Intended Audience :: Developers",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+    "Topic :: Software Development :: Testing",
+    "Topic :: Internet :: WWW/HTTP",
+    "License :: OSI Approved :: MIT License",
+]
+urls = {Homepage = "https://github.com/kevin1024/vcrpy"}
+requires-python = ">=3.8"
+dependencies = [
+    "PyYAML",
+    "wrapt",
+    "yarl",
+    # Support for urllib3 >=2 needs CPython >=3.10
```

Comment:
So what I don't like about this approach, is that it does not attempt to lock. If the non-locking is to preserve some old use-cases for python 3.8, then I suggest making those explicit.

This wild west, "go grab whatever versions of these packages!" is what caused the most breakages, and uncertainty, because it's harder for a dependency resolver to pick the right thing for all the packages.

Reply:
Let's leave this PR to migrate the config as is without changes, we can have a follow up PR to change the config

Reply:
What are the benefits of the PR? It declares the same things as the imperative code. Is that the benefit?

Reply:
Yes